packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -964,9 +975,32 @@ export class AttributesController extends ConsumptionBaseController {
             query["shareInfo.sourceAttribute"] = repositoryAttribute.id.toString();
             const sharedCopies = await this.getLocalAttributes(query);
 
-            ownSharedIdentityAttributeVersions.push(...sharedCopies);
+            ownSharedIdentityAttributePredecessors.push(...sharedCopies);
         }
 
-        return ownSharedIdentityAttributeVersions;
+        return ownSharedIdentityAttributePredecessors;
+    }
+
+    public async getSharedSuccessorsOfRepositoryAttribute(repositoryAttribute: LocalAttribute, query?: any): Promise<LocalAttribute[]> {
+        if (typeof query === "undefined") {
+            query = {};
+        }
```

Comment:
Create a deep copy of query if it's defined or else the function call will mutate the query object passed as an argument to the function. That's highly unidiomatic :-)

```ts
const query = { "content.@type": "GivenName" }
const successors = getSharedSuccessorsOfRepositoryAttribute(repoAttrId, query)

// query now has an extra 'shareInfo.sourceAttribute' key.
```

Same for the `getSharedPredecessorsOfRepositoryAttribute` method.